src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -411,38 +406,65 @@ private void ParseStatusLine(Span<byte> line, HttpResponseMessage response)
 
         private void ParseHeaderNameValue(Span<byte> line, HttpResponseMessage response)
         {
-            // TODO #23147: Use Span to get the header name and value rather than going through ValueStringBuilder
-
-            _sb.Clear();
             int pos = 0;
+            while (line[pos] != (byte)':' && line[pos] != (byte)' ')
```

Comment:
Should do https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanExtensions.cs

Probably need a reference to `System.Memory` though since they are extensions